consistent_web_main/main.go

```diff
@@ -171,7 +174,42 @@ func (main Main) processDelete(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusOK)
 }
 
+func recordLatency(latency time.Duration) {
+	fileMutex.Lock()
+	defer fileMutex.Unlock()
+	_, err := latencyFile.WriteString(fmt.Sprintf("%v\n", latency))
+	if err != nil {
+		fmt.Println("Error writing to file:", err)
+	}
+}
+
+func init() {
+	var err error
+	latencyFile, err = os.OpenFile("main_latencies.txt", os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	if err != nil {
+		fmt.Println("Error creating file:", err)
+		os.Exit(1)
+	}
+}
+
+func saveAndCloseFile() {
```

Comment:
I am confused about why we close and reopen the file for rewrites in this function. Also, can we make the function name more descriptive if we still want to reopen after closing?